File: shop/src/main/java/pl/lagodka/shop/model/dto/ProductDto.java
package pl.lagodka.shop.model.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.history.RevisionMetadata;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.PositiveOrZero;
import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ProductDto {
    private Long id;
    @NotBlank
    private String name;
    @NotNull
    @PositiveOrZero
    private Double price;
    @NotNull
    private boolean available;
    @PositiveOrZero
    private double quantity;

    private LocalDateTime createdDate;

    private String createdBy;

    private LocalDateTime lastModifiedDate;

    private String lastModifiedBy;

    private RevisionMetadata.RevisionType revisionType;

    private Integer revisionNumber;

    private String imageUrl;

}

File: java/shop/shop-front/src/api/models/product-dto.ts
/* tslint:disable */
/* eslint-disable */
export interface ProductDto {
  available: boolean;
  createdBy?: string;
  createdDate?: string;
  id?: number;
  imageUrl?: string;
  lastModifiedBy?: string;
  lastModifiedDate?: string;
  name: string;
  price: number;
  quantity?: number;
  revisionNumber?: number;
  revisionType?: 'UNKNOWN' | 'INSERT' | 'UPDATE' | 'DELETE';
}
